### PR TITLE
Add filter data for S3, EKS and DynamoDB

### PR DESCRIPTION
**Pull request by ChanochShayner**
To avoid rate limit errors,
adding a filter for resources by physical id to reduce the SDK calls.

### CODE REVIEW

File: cli/main.js [hunk 1]
```diff
@@ -177,6 +178,15 @@ async function main(opts) {
         sections = sections.filter(val => val); // reindex
     }
 
+    if (opts.resourcesArn) {
+        const rArn = opts.resourcesArn.split(",").map(x => x.toLowerCase())
```
Comment (achiar99) [inline cli/main.js]: why to lower case?

File: cli/main.js [hunk 2]
```diff
@@ -14,6 +14,7 @@ const _colors = require('colors');
 const pjson = require('./package.json');
 const { openStdin } = require("process");
 const CLI = true;
+const resourcesArn = []
```
Comment (achiar99) [inline cli/main.js]: maybe change to `physicalIdFilter` or something like that

File: js/datatables.js [hunk 3]
```diff
@@ -207,6 +208,67 @@ function lambdaRuntimeFormatter(data) {
...
+        return true
+    }
+}
+
+function isValidMethod(svc, method) {
+    if (svc === 'DynamoDB' && method.startsWith('describe')) return false
+    return true
+}
+
+function filterDataByArnIds(shouldFilter, data, svc, method) {
```
Comment (achiar99) [inline js/datatables.js]: shouldFilter is redundant

File: js/datatables.js [hunk 4]
```diff
@@ -207,6 +208,67 @@ function lambdaRuntimeFormatter(data) {
...
+    if (typeof(data[key]) === 'string') {
+        return resourcesArn.some(f => data[key].toLowerCase().indexOf(f) > -1)
+    } else {
+        return true
+    }
+}
+
+function isValidMethod(svc, method) {
+    if (svc === 'DynamoDB' && method.startsWith('describe')) return false
+    return true
```
Comment (achiar99) [inline js/datatables.js]:
```suggestion
    return !(svc === 'DynamoDB' && method.startsWith('describe'));
```

File: js/datatables.js [hunk 5]
```diff
@@ -207,6 +208,67 @@ function lambdaRuntimeFormatter(data) {
...
+
+function isKeyNameValidToFilter(key) {
+    return keysToFilterBy.some(substring => key.toLowerCase().includes(substring));
+}
+
+function isResourceEqualToArn(key, data) {
+    if (typeof(data[key]) === 'string') {
+        return resourcesArn.some(f => data[key].toLowerCase().indexOf(f) > -1)
+    } else {
+        return true
```
Comment (achiar99) [inline js/datatables.js:244]: why?

File: js/datatables.js [hunk 6]
```diff
@@ -301,7 +363,14 @@ function sdkcall(svc, method, params, alert_on_errors, backoff) {
                     reject(data);
                     return;
                 }
-                
+                console.log(`CloudFormationer log - sdkcall with svc: ${svc}, method: ${method}, params: ${params}`)
+                console.log('CloudFormationer log - sdkcall data before filtering: ', data)
```
Comment (achiar99) [inline js/datatables.js:370]: please log also the count of resources before the filter and after the filter

**Review by achiar99 — COMMENTED** [review]
Good job!
please see my comments
also, can we add some tests here?

File: cli/main.js (hunk 1 above)
Comment (ChanochShayner) [reply cli/main.js]: The resourcesArn is lowered as well

File: cli/main.js (hunk 2 above)
Comment (ChanochShayner) [reply cli/main.js]: done

File: js/datatables.js (hunk 3 above)
Comment (ChanochShayner) [reply js/datatables.js]: done

File: js/datatables.js (hunk 4 above)
Comment (ChanochShayner) [reply js/datatables.js]: done

File: js/datatables.js (hunk 5 above)
Comment (ChanochShayner) [reply js/datatables.js:244]: There are cases (for example in DynamoDB) that we are running on a `list` method but it's a list of objects that we don't need to filter - a list of tags related to a specific Dynamo table.

File: js/datatables.js (hunk 6 above)
Comment (ChanochShayner) [reply js/datatables.js:370]: It's pretty obvious the count from the current logs - 
And adding the count of the logs can be problematic as in every service the list is located in another place.
For DynamoDB:
image

For S3:
image

**Pull request merged**